CustomerSettingsActivity.java

```diff
@@ -0,0 +1,185 @@
+        mCustomerDatabase.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if(dataSnapshot.exists() && dataSnapshot.getChildrenCount()>0){
+                    Map<String, Object> map = (Map<String, Object>) dataSnapshot.getValue();
+                    if(map.get("name")!=null){
+                        mName = map.get("name").toString();
+                        mNameField.setText(mName);
+                    }
+                    if(map.get("phone")!=null){
+                        mPhone = map.get("phone").toString();
+                        mPhoneField.setText(mPhone);
+                    }
+                    if(map.get("profileImageUrl")!=null){
+                        mProfileImageUrl = map.get("profileImageUrl").toString();
+                        Glide.with(getApplication()).load(mProfileImageUrl).into(mProfileImage);
+                    }
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+
+
+    private void saveUserInformation() {
+        mName = mNameField.getText().toString();
+        mPhone = mPhoneField.getText().toString();
+
+        Map userInfo = new HashMap();
+        userInfo.put("name", mName);
+        userInfo.put("phone", mPhone);
+        mCustomerDatabase.updateChildren(userInfo);
+
+        if(resultUri != null) {
+
+            StorageReference filePath = FirebaseStorage.getInstance().getReference().child("profile_images").child(userID);
+            Bitmap bitmap = null;
+            try {
+                bitmap = MediaStore.Images.Media.getBitmap(getApplication().getContentResolver(), resultUri);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 20, baos);
+            byte[] data = baos.toByteArray();
+            UploadTask uploadTask = filePath.putBytes(data);
+
+            uploadTask.addOnFailureListener(new OnFailureListener() {
+                @Override
+                public void onFailure(@NonNull Exception e) {
+                    finish();
+                    return;
+                }
+            });
+            uploadTask.addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
+                @Override
+                public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
+                    Uri downloadUrl = taskSnapshot.getDownloadUrl();
+
+                    Map newImage = new HashMap();
+                    newImage.put("profileImageUrl", downloadUrl.toString());
+                    mCustomerDatabase.updateChildren(newImage);
+
+                    finish();
+                    return;
+                }
+            });
+        }else{

```

Comment:
To avoid such deep nesting you could invert the condition:
```
if (resultUri == null) {
  finish();
  return;
}
StorageReference filePath = ...
```